api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -370,6 +382,49 @@ private void streamToOutput(final boolean streamTimeoutReached) {
         }
     }
 
+    private void sendToDeadLetterQueue(final ConsumedEvent event, final int failedCommitsCount) {
+        // TODO: how do we handle AVRO here?
+
+        final String failedEventString = new String(event.getEvent(), StandardCharsets.UTF_8);
+        final JSONObject failedEvent = new JSONObject(failedEventString);
```

Comment:
IIUC, this is basically de-serializes the failed event.
Alternatively, the failed event could be stored as string (or byte array? relevant to my next comment).
For JSON type tha would involve only JSON-escaping that string, which should be more efficient than de-serialization, I think.

Reply:
You are right, but efficiency should not concern us for this use case, IMO. :)